scripts/esb_sniffer/extcap/esb_extcap.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (c) 2025 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+import argparse
+import signal
+import struct
+import sys
+
+PIPE_DATA = "/tmp/esb_sniffer_data"
+PIPE_COMM = "/tmp/esb_sniffer_comm"
+
+COMM_START_CAPTURE = "1"
+COMM_STOP_CAPTURE  = "2"
+
+def list_interfaces():
+    print("extcap {version=0.1}{display=ESB sniffer extcap}")
+    print("interface {value=ESBextcap}{display=ESB sniffer interface}")
+
+def list_dlts():
+    print("dlt {number=147}{name=ESB}{display=ESB}")
+
+def list_config():
+    return None
+
+def _pcap_header():
+    return struct.pack('@IHHiIII', 0xa1b2c3d4, 2, 4, 0, 0, 65535, 147)
+
+def capture(interface, ws_fifo, sn_pipe):
+    try:
+        ws = open(ws_fifo, 'wb')
+        ws.write(_pcap_header())
+        ws.flush()
+        while True:
+            pkt = sn_pipe.read(64)
+            if pkt:
+                ws.write(pkt)
+            else:
+                raise IOError
+
+            ws.flush()
+    except Exception:
+        ws.close()
```

Comment:
ws.close() is called in the exception. If the exception occurred because ws couldn't be opened, this will raise another exception